src/cdk/v2/destinations/reddit/transformV3.ts

```diff
@@ -0,0 +1,227 @@
+    )?.device?.advertisingId?.trim();
+  }
+  if (os === 'android') {
+    userData.aaid = (
+      message.context as { device?: { advertisingId?: string } }
+    )?.device?.advertisingId?.trim();
+  }
+  if (userData && hashData) {
+    userData.email = userData?.email ? hashSHA256(userData.email) : null;
+    userData.external_id = userData?.external_id ? hashSHA256(userData.external_id) : null;
+    userData.ip_address = userData?.ip_address ? hashSHA256(userData.ip_address) : null;
+    userData.phone_number = userData?.phone_number ? hashSHA256(userData.phone_number) : null;
+    userData.idfa = userData?.idfa ? hashSHA256(userData.idfa) : null;
+    userData.aaid = userData?.aaid ? hashSHA256(userData.aaid) : null;
+  }
+
+  return removeUndefinedAndNullValues(userData) as RedditUserData;
+};
+
+/**
+ * Prepares the event type by mapping the incoming event name to the destination's expected event type.
+ * @param message - The RudderMessage containing the event name.
+ * @param eventsMapping - Array of event mapping objects.
+ * @returns The mapped event type as a string, or undefined if not found.
+ */
+const prepareEventType = (
+  message: RudderMessage,
+  eventsMapping: Record<string, string>[],
+): RedditEventType | RedditEventType[] => {
+  const { event } = message;
+  if (!event) {
+    throw new InstrumentationError('Event name is required in the message');
+  }
+  const normalizedEvent = event.toLowerCase();
+  const eventsMap = getHashFromArrayWithDuplicate(eventsMapping);
+  const eventNames = new Set((eventsMap?.[normalizedEvent] as string[]) || []);
+
+  if (eventNames.size === 0) {
+    for (const ecomEventMap of ecomEventMaps) {
+      if (ecomEventMap.src.includes(normalizedEvent)) {
+        return { tracking_type: convertToUpperSnakeCase(ecomEventMap.dest) } as RedditEventType;
+      }
+    }
+  } else {
+    const eventTypes: RedditEventType[] = [];
+    eventNames.forEach((eventName: string) => {
+      eventTypes.push({
+        tracking_type: convertToUpperSnakeCase(eventName),
+      } as RedditEventType);
+    });
+    return eventTypes;
+  }
+
+  return { tracking_type: 'CUSTOM', custom_event_name: event };
+};
+
+const prepareProductsArrayWithItemCount = (message: RudderMessage): RedditEventMetadata => {
+  const { properties } = message;
+  if ((properties as { products: any[] })?.products?.length > 0) {
+    let itemCount = 0;
+    const products = (properties as { products: any[] })?.products.map((product) => {
+      itemCount += 1;
+      return {
+        id: product.product_id,
+        name: product.name,
+        category: product.category,
+      };
+    });
+    return {
+      item_count: itemCount,
+      products,
+    };
+  }
+
+  return {
+    item_count: 1,
+    products: [
+      {
+        id: (properties as { product_id: string })?.product_id,
+        name: (properties as { name: string })?.name,
+        category: (properties as { category: string })?.category,
+      },
```

Comment:
This piece of code has lot of type `as` assertion I would suggest you primitively add a type for products and then form it.

Also since the function has the return type the product array you should have the objects form in correct type